tests/unit/FileHelperTest.php

```diff
@@ -0,0 +1,325 @@
+
+        // Delete a nested folder
+        $filepath = vfsStream::url('root/folder_2/folder_3');
+        FilesHelper::delete_dir_with_files($filepath);
+        // And confirm the top level one still exists
+        $filepath = vfsStream::url('root/folder_2');
+        $expected = true;
+        $actual = is_dir( $filepath );
+        $this->assertEquals( $expected, $actual );
+    }
+
+    /**
+     * Test getListOfLocalFilesByDir method
+     *
+     * @return void
+     */
+    public function testGetListOfLocalFilesByDir() {
+        // Set up a virual folder structure
+        $structure = [
+            'folder_1' => [
+                'file_1.jpg' => 'file_1.jpg',
+                'file_2.jpg' => 'file_2.jpg',
+            ],
+            'folder_2' => [
+                'file_3.jpg' => 'file_3.jpg',
+                'file_4.jpg' => 'file_4.jpg',
+                'folder_3' => [
+                    'file_5.jpg' => 'file_5.jpg',
+                    'file_6.jpg' => 'file_6.jpg',
+                    
+                ],
+            ],
+        ];
+        $vfs = vfsStream::setup('root');
+        vfsStream::create($structure, $vfs);
+        // Set virtual WP root directory to /root/ for this test
+        $mock = Mockery::mock('overload:\WP2Static\SiteInfo');
+        $mock->shouldreceive('getPath')->andReturn(vfsStream::url('root') . "/");
+
+
+        // Top level folder
+        $filepath = vfsStream::url('root/folder_1');
+        $expected = ['/folder_1/file_1.jpg', '/folder_1/file_2.jpg'];
+        $actual = FilesHelper::getListOfLocalFilesByDir( $filepath );
+        $this->assertEquals( $expected, $actual );
+
+        // Nested folder
+        $filepath = vfsStream::url('root/folder_2/folder_3');
+        $expected = ['/folder_2/folder_3/file_5.jpg', '/folder_2/folder_3/file_6.jpg'];
+        $actual = FilesHelper::getListOfLocalFilesByDir( $filepath );
+        $this->assertEquals( $expected, $actual );
+
+        // Folder with subfolder
+        $filepath = vfsStream::url('root/folder_2');
+        $expected = [
+            '/folder_2/file_3.jpg',
+            '/folder_2/file_4.jpg',
+            '/folder_2/folder_3/file_5.jpg',
+            '/folder_2/folder_3/file_6.jpg'
+        ];
+        $actual = FilesHelper::getListOfLocalFilesByDir( $filepath );
+        $this->assertEquals( $expected, $actual );
+    }
+
+    /**
+     * Test filePathLooksCrawlable method
+     *
+     * @return void
+     */
+    public function testFilePathLooksCrawlable() {
+        // Default accepted extension
+        $expected = true;
+        $actual = FilesHelper::filePathLooksCrawlable( "/path/to/foo.jpg" );
+        $this->assertEquals( $expected, $actual );
+
+        // Default disallowed extension
+        $expected = false;
+        $actual = FilesHelper::filePathLooksCrawlable( "/path/to/foo.txt" );
+        $this->assertEquals( $expected, $actual );
+
+        // Default disallowed filename
```

Comment:
Duplicate of above test? Can we load a whole array in to test or just use simple loop to cover all extension types and all disallowed filenames? They are fairly static and will increase coverage.

Reply:
so, looking further down, next test appears to do what I was expecting here. Cannot see the difference between the two tests, may as well combine or rename those functions if there's a clear difference